packages/components/src/components/rating/rating.jsx

```diff
@@ -0,0 +1,134 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import Text from '../text';
+
+const Rating = ({
+    allow_half_rating = false,
+    background_class,
+    direction = 'ltr',
+    disable_hover = false,
+    display_default_content,
+    element_class,
+    icon_selected,
+    icon_unselected,
+    max_rating = 5,
+    onClick,
+    readonly = false,
+    show_rating = false,
+    value,
+}) => {
+    const [enabled_rate_index, setEnabledRateIndex] = React.useState(-1);
+    const [hovered_rate_index, setHoveredRateIndex] = React.useState(-1);
+    const [is_hover_enabled, setIsHoverEnabled] = React.useState(false);
+    const ratingContainerRef = React.useRef(null);
+
+    const calculateRating = e => {
+        const { width, left, right } = ratingContainerRef.current.getBoundingClientRect();
+        const rating_start_element = direction === 'ltr' ? e.clientX - left : right - e.clientX;
+        const selected_rating = (rating_start_element / width) * max_rating;
+        if (selected_rating > 0.4) {
+            const calculated_rating = allow_half_rating
+                ? Math.round((selected_rating + 0.5 / 2) / 0.5) * 0.5 // This will always make sure that, if we hover between two ratings, it will select the nearest number
+                : Math.round(selected_rating);
+            return calculated_rating;
+        }
+        return 0;
+    };
+
+    const handleClick = e => {
+        if (readonly) {
+            return;
+        }
+        setIsHoverEnabled(false);
+        const rating_value = calculateRating(e);
+        setEnabledRateIndex(rating_value);
+        onClick(rating_value);
+    };
+
+    const handleMouseMove = e => {
+        if (readonly || disable_hover) {
+            return;
+        }
+        setIsHoverEnabled(true);
+        setHoveredRateIndex(calculateRating(e));
+    };
+
+    const handleMouseLeave = () => {
+        if (readonly || disable_hover) {
+            return;
+        }
+        setHoveredRateIndex(-1); // Reset to default state
+        setIsHoverEnabled(false);
+    };
+
+    const validateInitialValue = () => {
+        if (value < 0 || value > max_rating) {
+            setEnabledRateIndex(0);
+        } else {
+            const set_value = allow_half_rating ? parseFloat(value) : parseInt(value);
+            setEnabledRateIndex(set_value);
```

Comment:
```suggestion
            setEnabledRateIndex(parseFloat(value));
```

Reply:
If rating is done is to be provided in whole numbers, it should not be converted to float. Hence based on the `allow_half_rating` prop value, I am parsing it as an integer or float